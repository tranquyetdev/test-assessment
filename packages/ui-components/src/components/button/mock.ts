import { ButtonProps } from './button';

export const mockButtonProps: Record<ButtonProps['type'], ButtonProps> = {
  button: {
    type: 'button',
    children: 'button',
    onClick: () => console.log('hello world'),
  },
  link: {
    type: 'link',
    children: 'link',
    href: 'https://github.com/',
    title: 'Github',
  },
};
