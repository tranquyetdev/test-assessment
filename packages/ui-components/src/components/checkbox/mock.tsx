import { CheckBoxProps, defaultBgcolor } from '.';

export const mockCheckBoxProps: CheckBoxProps = {
  label: 'Developer',
  bgColor: defaultBgcolor,
  id: 'checkbox',
  checked: false,
  disabled: false,
};
