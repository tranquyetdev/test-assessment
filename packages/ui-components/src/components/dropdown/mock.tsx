export const mockDropdownProps = {
  content: (
    <ul className="menu">
      <li className="menu-item">
        <button>Menu 1</button>
      </li>
      <li className="menu-item">
        <button>Menu 2</button>
      </li>
    </ul>
  ),
  button: <div className="border">Show content</div>,
};
