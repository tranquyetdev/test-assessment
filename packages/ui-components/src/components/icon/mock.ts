import { IconName, IconProps } from ".";

export const mockIcons: IconProps & { name?: IconName }[] = [
  { name: "arrow-left" },
  { name: "arrow-right" },
  { name: "check" },
  { name: "chevron-down" },
  { name: "remove" },
  { name: "edit" },
  { name: "notification" },
  { name: "plus" },
  { name: "publish" },
  { name: "qa" },
  { name: "refresh" },
  { name: "save" },
  { name: "search" },
  { name: "caution" },
  { name: "eye-hide" },
  { name: "eye-show" },
]