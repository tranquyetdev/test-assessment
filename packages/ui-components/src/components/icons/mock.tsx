import { IconName, IconProps } from ".";

export const mockIcons: IconProps & { name?: IconName }[] = [
  { name: "arrow-left" },
  { name: "arrow-right" },
  { name: "check" },
  { name: "chevron-down" },
  { name: "edit" },
  { name: "notification" },
  { name: "plus" },
  { name: "publish" },
  { name: "qa" },
  { name: "refresh" },
  { name: "remove" },
  { name: "save" },
  { name: "caution" },
  { name: "eye-hide" },
  { name: "eye-show" },
]