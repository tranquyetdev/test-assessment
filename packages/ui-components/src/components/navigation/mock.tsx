import { NavigationProps } from './types';

export const mockNavigationProps: NavigationProps = {
  menuItems: [
    {
      title: 'Test Management',
      url: '/test-management',
    },
    {
      title: 'Candidate',
      url: '/candidate',
    },
    {
      title: 'Candidate',
      url: '/candidate',
    },
  ],
  currentPathname: '/test-management',
};
