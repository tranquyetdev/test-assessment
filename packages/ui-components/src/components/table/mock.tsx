import { TableProps } from '.';

export const mockTableProps: TableProps = {
  rows: [
    {
      name: 'name',
      published: new Date(),
      sent: 1,
      submitted: 1,
      author: 'author',
      position: 'Developer',
      level: 'level',
    },
  ],
  columns: [
    {
      title: 'Test name',
      render: 'name',
    },
    {
      title: 'Test status',
      render: (row) => <span>{row.published ? 'Published' : 'Draft'}</span>,
    },
    {
      title: 'Sent',
      render: 'sent',
    },
    {
      title: 'Submitted',
      render: 'submitted',
    },
    {
      title: 'Author',
      render: 'author',
    },
    {
      title: 'Job position',
      render: 'position',
    },
    {
      title: 'Level',
      render: 'level',
    },
    {
      title: 'Action',
      render: () => <>Button</>,
    },
  ],
};
