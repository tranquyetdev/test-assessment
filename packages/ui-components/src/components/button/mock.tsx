import { RenderIcon } from '../icons';
import { ButtonProps } from './button';

export const mockButtonProps: Record<ButtonProps['type'], ButtonProps> = {
  button: {
    type: 'button',
    children: 'button',
    onClick: () => alert('hello world'),
    LeftIcon: <RenderIcon name="caution" className="text-error" />,
    RightIcon: (
      <RenderIcon name="caution" className="text-error" />
    ),
  },
  link: {
    type: 'link',
    children: 'link',
    href: 'https://github.com/',
    title: 'Github',
    LeftIcon: <RenderIcon name="caution" className="text-error" />,
    RightIcon: (
      <RenderIcon name="caution" className="text-error" />
    ),
  },
};
