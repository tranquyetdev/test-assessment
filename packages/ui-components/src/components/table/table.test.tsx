import React from 'react';
import { render } from '@testing-library/react';

import { Table } from './';
import { mockTableProps } from './mock';

describe('Table', () => {
  it(`should render successfully`, () => {
    const table = render(<Table {...mockTableProps} />);
    expect(table).toMatchSnapshot();
  });
});
