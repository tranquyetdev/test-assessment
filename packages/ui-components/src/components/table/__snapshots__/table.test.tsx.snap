// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render successfully 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="border border-neutral-border"
      >
        <thead>
          <tr>
            <th
              class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
            >
              Test name
            </th>
            <th
              class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
            >
              Test status
            </th>
            <th
              class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
            >
              Sent
            </th>
            <th
              class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
            >
              Submitted
            </th>
            <th
              class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
            >
              Author
            </th>
            <th
              class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
            >
              Job position
            </th>
            <th
              class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
            >
              Level
            </th>
            <th
              class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
            >
              Action
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="text-subhead-2 py-4 px-3"
            >
              name
            </td>
            <td
              class="text-subhead-2 py-4 px-3"
            >
              <span>
                Published
              </span>
            </td>
            <td
              class="text-subhead-2 py-4 px-3"
            >
              1
            </td>
            <td
              class="text-subhead-2 py-4 px-3"
            >
              1
            </td>
            <td
              class="text-subhead-2 py-4 px-3"
            >
              author
            </td>
            <td
              class="text-subhead-2 py-4 px-3"
            >
              Developer
            </td>
            <td
              class="text-subhead-2 py-4 px-3"
            >
              level
            </td>
            <td
              class="text-subhead-2 py-4 px-3"
            >
              Button
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="border border-neutral-border"
    >
      <thead>
        <tr>
          <th
            class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
          >
            Test name
          </th>
          <th
            class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
          >
            Test status
          </th>
          <th
            class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
          >
            Sent
          </th>
          <th
            class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
          >
            Submitted
          </th>
          <th
            class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
          >
            Author
          </th>
          <th
            class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
          >
            Job position
          </th>
          <th
            class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
          >
            Level
          </th>
          <th
            class="text-subhead-2 py-2 px-3 text-neutral-placeholder bg-neutral-divider"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="text-subhead-2 py-4 px-3"
          >
            name
          </td>
          <td
            class="text-subhead-2 py-4 px-3"
          >
            <span>
              Published
            </span>
          </td>
          <td
            class="text-subhead-2 py-4 px-3"
          >
            1
          </td>
          <td
            class="text-subhead-2 py-4 px-3"
          >
            1
          </td>
          <td
            class="text-subhead-2 py-4 px-3"
          >
            author
          </td>
          <td
            class="text-subhead-2 py-4 px-3"
          >
            Developer
          </td>
          <td
            class="text-subhead-2 py-4 px-3"
          >
            level
          </td>
          <td
            class="text-subhead-2 py-4 px-3"
          >
            Button
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
